pkg/event/target/mqtt.go

```diff
@@ -96,7 +101,8 @@ func (target *MQTTTarget) Send(eventData event.Event) error {
 
 	token := target.client.Publish(target.args.Topic, target.args.QoS, false, string(data))
 
-	if token.Wait() {
+	// In-case if the mqtt broker is offline, the token.Wait() waits indefinitely
+	if target.args.QueueDir == "" && token.Wait() {
```

Comment:
Basically you need to do below to handle all cases.

```go
if target.args.QueueDir == "" {
    // Wait indefinitely for publish to complete.
    token.Wait()
}

return token.Error()
```

Reply:
yes. done